### PR TITLE
Abstract cmd valid args get behind the factory

### PR DESCRIPTION
**Pull request by niuzhenguo**
**What this PR does / why we need it**:
This abstract retrieving the list of handled resources for valid args as a function to follow more conventions.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/kubectl/cmd/annotate.go [hunk 1]
```diff
@@ -101,16 +100,8 @@ var (
...
-	p, err := f.Printer(nil, printers.PrintOptions{
-		ColumnLabels: []string{},
-	})
-	cmdutil.CheckErr(err)
-	if p != nil {
-		validArgs = p.HandledResources()
-		argAliases = kubectl.ResourceAliases(validArgs)
-	}
+	validArgs := []string{}
+	validArgs = cmdutil.ValidArgList(f)
```
Comment (dixudx) [inline pkg/kubectl/cmd/annotate.go]: Just `validArgs := cmdutil.ValidArgList(f)` is enough.

Same as below.

Comment (niuzhenguo) [reply pkg/kubectl/cmd/annotate.go]: yes, thanks! will address that soon

**Comment by niuzhenguo** [comment]
@dixudx update according to your comments.

**Comment by mengqiy** [comment]
LGTM
Please squash commits.

**Comment by niuzhenguo** [comment]
@mengqiy Done

**Comment by dixudx** [comment]
@niuzhenguo Good cleanup. 👍

/lgtm

File: pkg/kubectl/cmd/util/printing.go [hunk 2]
```diff
@@ -223,3 +223,17 @@ func ValidResourceTypeList(f ClientAccessFactory) string {
 	
 	`)
 }
+
+// Retrieve a list of handled resources from printer as valid args
```
Comment (liggitt) [inline pkg/kubectl/cmd/util/printing.go]: keep the `TODO: this needs to be abstracted behind the factory like ValidResourceTypeList and use discovery`... this change removes duplicate code, but using a statically compiled-in list is still wrong

**Comment by liggitt** [comment]
/approve 
this cleans up duplication, but building these using compiled-in lists means the results are incorrect. keep the TODO to switch this to use discovery

File: pkg/kubectl/cmd/util/printing.go (hunk 2 above)
Comment (niuzhenguo) [reply pkg/kubectl/cmd/util/printing.go]: thanks @liggitt , will add the TODO back soon!

**Comment by niuzhenguo** [comment]
@liggitt Added the TODO back as  your comments, can you pleas help to approve this again? thanks!

**Comment by mengqiy** [comment]
/lgtm
You may want to link a issue to make the bot happy.

**Pull request merged**